src/Hazelcast.Net/Clustering/RetryStrategy.cs

```diff
@@ -71,19 +72,39 @@ public RetryStrategy(string action, int initialBackOffMilliseconds, int maxBackO
             _action = action.ToLowerInvariant();
 #pragma warning restore CA1308
 
-
+            if (initialBackOffMilliseconds < 0) throw new ConfigurationException("Initial back-off must be greater than or equal to zero.");
             _initialBackoffMilliseconds = initialBackOffMilliseconds;
+            if (maxBackOffMilliseconds < 0) throw new ConfigurationException("Maximum back-off must be greater than or equal to zero.");
             _maxBackoffMilliseconds = maxBackOffMilliseconds;
+            if (multiplier <= 0) throw new ConfigurationException("Multiplier must be greater than zero.");
             _multiplier = multiplier;
             _timeoutMilliseconds = timeoutMilliseconds;
+            if (jitter < 0 || jitter > 1) throw new ConfigurationException("Jitter must be between zero and one, inclusive.");
             _jitter = jitter;
-            _currentBackOffMilliseconds = initialBackOffMilliseconds;
 
             _logger = loggerFactory?.CreateLogger<RetryStrategy>() ?? throw new ArgumentNullException(nameof(loggerFactory));
 
             Restart();
         }
 
+        /// <summary>
+        /// (internal for tests only) Gets the delay.
+        /// </summary>
+        internal int GetDelay(int elapsed)
+        {
+            var delay = (int)(_currentBackOffMilliseconds * (1 - _jitter * (1 - RandomProvider.NextDouble())));
+            if (_timeoutMilliseconds >= 0) delay = Math.Min(delay, Math.Max(0, (int)(_timeoutMilliseconds - elapsed)));
+            return delay;
+        }
+
+        /// <summary>
+        /// (internal for tests only) Gets the new back-off.
+        /// </summary>
```

Comment:
We are using it in `WaitAsync` indeed. If we were only using it there, it would be `private`. We are making it `internal` so we can also use it in tests. But it's `internal` only for tests, it should not be used otherwise.